lips/lip-6.md

```diff
@@ -0,0 +1,79 @@
+* *Validator Operator* ("operator") is an entity that manages its on-chain Libra account and runs a Validator Node on behalf of the Validator Owner who designated this operator to be running the node. There can be one Validator Operator account on-chain that runs two Validator Nodes for two Validator Owners, one node per one owner. It is not allowed to run a single Validator Node per multiple Validator Owners even when the nodes are run by the same Validator Operator. The Validator Operator has the permission to rotate the Validator Configs for its Validator Owners.
+
+# Specification
+## Overview
+Each Validator Owner holds a Libra account with a *Validator Owner role* (see [lip-2: Libra Roles and Permissions](https://github.com/libra/lip/blob/master/lips/lip-2.md)). A Validator Owner designates a Validator Operator to run a Validator Node
+A Validator Owner designates a single Validator Operator account to run a Validator Node. Validator Operator is a Libra account with a *Validator Operator role*. Each Validator Operator may run nodes for multiple Validator Owners, therefore there is a one-to-many relationship between operator accounts and owner accounts. If an operator OP_1 is designated to run Validator Nodes for two Validator Owners V_1 and V_2, the operator must run two distinct instances of Validator Nodes.
+
+Each Validator Owner account stores a consensus configuration that consists of the consensus public key, the networking information (for the Validator Node to connect to other Validator Nodes) and the Full Node networking information (for the public Full Nodes and clients and, under certain circumstances, for other Validator Nodes themselves to connect to this Validator Node). The Validator Operator designated by the Validator Owner is responsible for running a node with this consensus config and is responsible for updating/rotating this config as needed.
+
+The Validator Set is a map of Validator Owner addresses to a copy of their consensus configs. Each Validator Owner address appears in the Validator Set at most once. This set is stored under the LibraRoot address and reflects the configuration that currently runs consensus protocol. Any change to this set defines a new epoch by triggering a new epoch event. If a transaction causes an epoch change, that transaction is the last transaction committed in the current epoch — any subsequent transactions in that block or future blocks from that epoch will be ignored. Once the transaction has been committed, the new Validator Set can start the next epoch of the consensus protocol.
```

Comment:
I think we need a separate discussion of configs.  If someone were trying to re-implement this, it would not be sufficient just to publish the validator set at LibraRoot any way they want (as I understand it) -- clients look at libra configs directly, right?

The discussion of reconfiguration is really important, and probably should appear separate and earlier.

So, yeah, we need a detailed discussion of configs either in this document or elsewhere.  Should that talk about other configs?  Much of the discussion would be about validator config, because it's the hardest.